src/plat/spike/config.cmake

```diff
@@ -30,6 +30,7 @@ if(KernelPlatformSpike)
     declare_default_headers(
         TIMER_FREQUENCY 10000000 PLIC_MAX_NUM_INT 0
         INTERRUPT_CONTROLLER drivers/irq/riscv_plic_dummy.h
+        CLINT_MTIME_OFFSET 0xbff8
```

Comment:
This offset seems to be the same for every CLINT supported, wouldn't it also work as a #define nearer to `riscv_read_time`?

Reply:
Yes I can do that instead, but is it simpler? That is what I had initially but I thought it would be better to have it in `config.cmake` given it is a platform-specific value rather than the kernel source code.

Reply:
> This offset seems to be the same for every CLINT supported, wouldn't it also work as a #define nearer to `riscv_read_time`?

I think it's because all the current platforms just copy the same SiFive CLINT. If we make platform state this in their DTB, there could be less redundancy,

Reply:
I was told that the memory layout of the SiFive CLINT is not guaranteed to be the same.

Reply:
Oh, do you mean have the mtime offset as a field in the CLINT DTS node or something?

Reply:
Another platform-specific way would be to define a CLINT driver that can bind to all of the platforms with the matching compatibility string and it implements `riscv_read_time` and knows the register offset. But I think if every platform we support has the same CLINT with the same offset then it's more pragmatic to just have less abstraction and move the definitions closer to the code that's using them.